Rust/Rust.sublime-syntax

```diff
@@ -187,20 +176,20 @@ contexts:
 
     - match: '\b(const)\s+({{identifier}})'
       captures:
-        1: storage.type.rust
+        1: keyword.declaration.rust
         2: entity.name.constant.rust
       push: declaration-after-identifier
 
     - match: '\b(static)\s+(?:(mut)\s+)?({{identifier}})'
       captures:
-        1: storage.type.rust
+        1: keyword.declaration.rust
         2: storage.modifier.rust
         3: entity.name.constant.rust
       push: declaration-after-identifier
 
     - match: \b(let)\s+(?:(mut)\s+)?
       captures:
-        1: storage.type.rust
+        1: keyword.declaration.rust
```

Comment:
Do we not have a subscope for `let`? I think Golang uses `keyword.declaration.var` for example

Reply:
I'm not aware of a language-agnostic subscope for variables, but I also didn't dig too deep into. For the sake of not overspecializing, I chose the defensive "no subscope" approach. I'll see if I can find some existing scope names or RFCs where this was discussed before. (I wish we had some properly formulated, agreed upon and indexed scope naming guidelines …)

Reply:
Java scopes `var` `storage.type.variant` as this is what it semantically is, next to strict types such as `int` or `boolean`.

A possibly agnostic scope for such keywords would be `keyword.declaration.variable` as it is what those keywords most likely do.

Same like `keyword.declaration.function` being used for `def` in python for instance.

Reply:
`variable` was also my initial guess and if I'm not alone with that, that is enough of an argument for me to start using it. I agree that it doesn't apply to Java because the `var` there indicates type inference and not the declaration of a variable.

I also added the sub-scopes `static` and `constant` for the other two variable-esque declaration keywords `static` and `const`.